src/api.rs

```diff
@@ -14,6 +15,42 @@ pub enum DriverType {
     Xen,
 }
 
+/// Supports passing initialization parameters to the driver
+///
+/// Some drivers can support optional extra initialization parameters.
+///
+/// This is required to initialize the KVM driver, which needs a `domain_name` and
+/// a `kvm_socket` parameters.
+///
+/// This is equivalent to LibVMI's `vmi_init_data_type_t`
+///
+/// # Examples
+///
+/// ```
+///let kvmi_socket = DriverInitParam::KVMiSocket(String::from("/tmp/introspector"));
+///let mut drv = microvmi::init(domain_name, DriverType::KVM, Some(kvmi_socket));
+/// ```
+#[derive(Debug)]
+pub enum DriverInitParam {
+    KVMiSocket(String),
+}
+
+impl From<DriverInitParamFFI> for DriverInitParam {
```

Comment:
@rageagainsthepc I tried to follow your advice and added another enum with a CString instead, and a From trait implementation (should I use Into) ?

It doesn't compile right now because there is a move, and I don't know how to solve it
![Capture d’écran de 2020-08-09 17-16-29](https://user-images.githubusercontent.com/964610/89735608-1a3cd300-da64-11ea-9f03-f00b30d005a2.png)

if you have any idea.

Reply:
`From` is usually better than `Into` because you get the `Into` implementation for free IIRC. However since for converion is fallible you should perhaps consider using `TryFrom` instead.